src/AbstractAggregateDefinition.php

```diff
@@ -81,7 +81,7 @@ public function metadataMatcher(string $aggregateId, int $aggregateVersion): ?Me
             ->withMetadataMatch('_aggregate_version', Operator::GREATER_THAN_EQUALS(), $aggregateVersion);
     }
 
-    public function metadataEnricher(string $aggregateId, int $aggregateVersion): ?MetadataEnricher
+    public function metadataEnricher(string $aggregateId, int $aggregateVersion, Message $causation = null): ?MetadataEnricher
```

Comment:
I added the `$causation` param, so we can add things like `causation_id` and `causation_name`.